src/main/java/ca/on/oicr/pineryreports/reports/impl/ProjectSequencingReport.java

```diff
@@ -0,0 +1,212 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.ParseException;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pineryreports.data.ColumnDefinition;
+import ca.on.oicr.pineryreports.reports.TableReport;
+import ca.on.oicr.ws.dto.RunDto;
+import ca.on.oicr.ws.dto.RunDtoPosition;
+import ca.on.oicr.ws.dto.RunDtoSample;
+import ca.on.oicr.ws.dto.SampleDto;
+import ca.on.oicr.ws.dto.UserDto;
+
+import com.google.common.base.Functions;
+import com.google.common.collect.Sets;
+
+public class ProjectSequencingReport extends TableReport {
+  
+  private static class ReportObject {
+    private final SampleDto dilution;
+    private final RunDto run;
+    private final RunDtoPosition lane;
+    private final UserDto poolCreator;
+    
+    public ReportObject(SampleDto dilution, RunDto run, RunDtoPosition lane, UserDto poolCreator) {
+      this.dilution = dilution;
+      this.run = run;
+      this.lane = lane;
+      this.poolCreator = poolCreator;
+    }
+
+    public SampleDto getDilution() {
+      return dilution;
+    }
+
+    public RunDto getRun() {
+      return run;
+    }
+
+    public RunDtoPosition getLane() {
+      return lane;
+    }
+    
+    public UserDto getPoolCreator() {
+      return poolCreator;
+    }
+  }
+  
+  public static final String REPORT_NAME = "sequencing";
+  private static final String OPT_PROJECT = "project";
+  
+  public static final String GSLE_USER = "Geospiza";
+
+  private static final List<ColumnDefinition> COLUMNS = Collections.unmodifiableList(Arrays.asList(
+      new ColumnDefinition("Stock"),
+      new ColumnDefinition("Pool"),
+      new ColumnDefinition("Pool Creator"),
+      new ColumnDefinition("Pool Created"),
+      new ColumnDefinition("Dilutions"),
+      new ColumnDefinition("Library"),
+      new ColumnDefinition("Run"),
+      new ColumnDefinition("Lane"),
+      new ColumnDefinition("Index")
+  ));
+  
+  private String project;
+  Map<String, SampleDto> allSamplesById;
+  List<ReportObject> reportData;
+  
+  @Override
+  public String getReportName() {
+    return REPORT_NAME;
+  }
+
+  @Override
+  public Collection<Option> getOptions() {
+    Set<Option> opts = Sets.newHashSet();
+    
+    opts.add(Option.builder()
+        .longOpt(OPT_PROJECT)
+        .hasArg()
+        .argName("code")
+        .required()
+        .desc("Project to report on (required)")
+        .build());
+    
+    return opts;
+  }
+
+  @Override
+  public void processOptions(CommandLine cmd) throws ParseException {
+    this.project = cmd.getOptionValue(OPT_PROJECT);
+  }
+
+  @Override
+  public String getTitle() {
+    return project + " Sequencing Report";
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 29 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)